haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -175,18 +187,31 @@ def __init__(self, retriever: "EmbeddingRetriever"):
                 f"This can be set when initializing the DocumentStore"
             )
 
-    def embed(self, texts: Union[List[List[str]], List[str], str]) -> List[np.ndarray]:
+    def embed(self, texts: Union[List[List[str]], List[str], str]) -> np.ndarray:
         # texts can be a list of strings or a list of [title, text]
         # get back list of numpy embedding vectors
-        emb = self.embedding_model.encode(texts, batch_size=self.batch_size, show_progress_bar=self.show_progress_bar)
-        emb = [r for r in emb]
+        emb = self.embedding_model.encode(
+            texts, batch_size=self.batch_size, show_progress_bar=self.show_progress_bar, convert_to_numpy=True
+        )
         return emb
 
-    def embed_queries(self, texts: List[str]) -> List[np.ndarray]:
-        return self.embed(texts)
+    def embed_queries(self, queries: List[str]) -> np.ndarray:
+        """
+        Create embeddings for a list of queries.
 
-    def embed_documents(self, docs: List[Document]) -> List[np.ndarray]:
-        passages = [[d.meta["name"] if d.meta and "name" in d.meta else "", d.content] for d in docs]  # type: ignore
+        :param queries: List of queries to embed.
+        :return: Embeddings, one per input query, shape: (queries, embedding_dim)
+        """
+        return self.embed(queries)
+
+    def embed_documents(self, docs: List[Document]) -> np.ndarray:
+        """
+        Create embeddings for a list of documents.
+
+        :param docs: List of documents to embed.
+        :return: Embeddings, one per input document, shape: (documents, embedding_dim)
+        """
+        passages = [[d.meta["name"] if d.meta and "name" in d.meta else "", d.content] for d in docs]
```

Comment:
Wow this is a big assumption about the `name` field in `meta`! I wasn't aware it was embedded by the retriever this way, i assumed it was on the docstore to decide what should be embedded and what not. And it's not even configurable. Maybe even worth highlighting to the documentation team?

Reply:
I agree, that's weird. Best would be to change it as the functionality to include meta fields is clearly the responsibility of `EmbeddingRetriever` and not its encoders. For clarity and transparency, I would change that in another PR (so it's also mentioned in the release notes).